src/app-layer-ssl.c

```diff
@@ -154,10 +155,12 @@ enum SslConfigEncryptHandling {
 
 typedef struct SslConfig_ {
     enum SslConfigEncryptHandling encrypt_mode;
-    /** dynamic setting for ja3: can be enabled on demand if not explicitly
-     *  disabled. */
+    /** dynamic setting for ja3 and ja4: can be enabled on demand if not
+     *  explicitly disabled. */
     SC_ATOMIC_DECLARE(int, enable_ja3);
     bool disable_ja3; /**< ja3 explicitly disabled. Don't enable on demand. */
+    SC_ATOMIC_DECLARE(int, enable_ja4);
+    bool disable_ja4; /**< ja4 explicitly disabled. Don't enable on demand. */
```

Comment:
Why do we need 2 ? enable and disable ?..

Reply:
Looks like `disable_ja4` is never set

Reply:
I'm trying to remember why we had this logic in the first place.  I think it had to do with nss being optional, but as of 7 all builds of Suricata have hashing available, so maybe we should get rid of this logic.  Were there other reasons @victorjulien ?

Reply:
I'd think it is split into separate enable/disable because JA3 support can be configured to be auto-enabled if a rule is parsed that needs it, explicitly disabled. I will have to take another look at how it's used to match JA3's behaviour.

Reply:
In the end, use it or do not put it ;-)

Reply:
Used. :)